### ISSUE TITLE
Snap Events

### ISSUE THREAD

**Issue by DavMila** [mozilla/standards-positions#1008]
## Request for Mozilla Position on an Emerging Web Specification

* Specification title: Snap Events
* Specification or proposal URL (if available): https://drafts.csswg.org/css-scroll-snap-2/#snap-events
* Explainer URL (if available): Snapchanged: https://github.com/argyleink/ScrollSnapExplainers/tree/main/js-snapChanged
Snapchanging: https://github.com/argyleink/ScrollSnapExplainers/tree/main/js-snapChanging

**Comment by emilio** [comment]
cc @hiikezoe. I think generally this is fine, the integration looks very similar to scroll events.

**Comment by hiikezoe** [comment]
Yeah, the basic idea is pretty reasonable to me.  Though I don't quite understand why SnapEvent can have only one snapTargetBlock and snapTargetInline on each axis respectively. I support it should be a List of nodes. Anyways it should be discussed in CSSWG not here.

**Comment by flackr** [comment]
> Yeah, the basic idea is pretty reasonable to me. Though I don't quite understand why SnapEvent can have only one snapTargetBlock and snapTargetInline on each axis respectively. I support it should be a List of nodes. Anyways it should be discussed in CSSWG not here.

This was in fact was discussed in the CSSWG in https://github.com/w3c/csswg-drafts/issues/9622

The TLDR is that when following an element after a change in layout browsers must have chosen some element to follow per https://drafts.csswg.org/css-scroll-snap-1/#re-snap so we have standarized in that issue how to choose which element is chosen (when multiple are aligned) and are exposing that chosen element in the event.

**Comment by hiikezoe** [comment]
Thanks for the pointer! I haven't followed the spec issue yet indeed. :)
